### PR TITLE
Support opening and comparing versions in viewer

### LINKED ISSUE #39139: Open versions of a file by clicking on the version list item

### LINKED ISSUE #39140: Compare 2 versions of a file
Provide an option to compare 2 versions of a file by marking them and opening both side-by-side in some kind of a "dual-pane", so the user can scroll both to visually compare them.

### PR DESCRIPTION
**Pull request by artonge**
Fix #39139
Fix #39140 

Tests will be added in text as there is no default viewer handler that covers version comparison support for now

### Clicking the version just opens a single preview
Screenshot 2023-08-03 at 15 29 51

### Comparing menu
- Comparing is available for any but the current version
Screenshot 2023-08-03 at 15 27 46

- Clicking it will compare the picked with the current version
Screenshot 2023-08-03 at 15 27 50

### Todo

- [x] https://github.com/nextcloud/viewer/pull/1782
- [x] https://github.com/nextcloud/viewer/pull/1835
- [x] BUG: Filter out image/video files
- [x] BUG: Currently comparing when no viewer is open is not working
- [x] Add screencasts
- [x] Tested with:
  - [x] Supporting version comparison
    - [x] Text file
        - [x] https://github.com/nextcloud/viewer/pull/1806
        - [x] https://github.com/nextcloud/text/pull/2832
    - [x] PDF
      - [x] https://github.com/nextcloud/files_pdfviewer/pull/791
  - [x] No version preview
    - [x] Images
    - [x] Video
    - [x] Music
    - [x] Files types not supported by the viewer

### CODE REVIEW

File: apps/files_versions/src/views/VersionTab.vue [hunk 1]
```diff
@@ -78,6 +86,37 @@ export default {
...
+				basename: this.fileInfo.name,
+				filename: this.fileInfo.path + this.fileInfo.name,
+				permissions: davPermissions,
+				fileid: this.fileInfo.id,
+			}
+		},
+
+		/** @return {boolean} */
+		canView() {
+			return window.OCA?.Viewer.mimetypesCompare.includes(this.fileInfo.mimetype)
```
Comment (juliushaertl) [inline apps/files_versions/src/views/VersionTab.vue:114]:
```suggestion
			return window.OCA?.Viewer.mimetypesCompare?.includes(this.fileInfo.mimetype)
```

To merge separately

**Review by juliusknorr — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Review by max-nextcloud — APPROVED** [review]

**Comment by juliushaertl** [comment]
/backport 5902325e235d90bbaec9d2f152f8cf453f197c59 to stable27

**Comment by AndyScherzinger** [comment]
Backport bot is no friend of mine nor of @juliushaertl 😭

**Pull request merged**